loadflow/loadflow-api/src/main/java/com/powsybl/loadflow/LoadFlowProvider.java

```diff
@@ -43,15 +47,22 @@
      * to be stateless so that it can be call simultaneously with different arguments (a different network for instance)
      * without any concurrency issue.
      *
-     * @param network the network
+     * @param network            the network
      * @param computationManager a computation manager to external program execution
-     * @param workingVariantId variant id of the network
-     * @param parameters load flow execution parameters
-     * @param reporter the reporter used for functional logs
+     * @param workingVariantId   variant id of the network
+     * @param parameters         load flow execution parameters
+     * @param reporter           the reporter used for functional logs
      * @return a {@link CompletableFuture} on {@link LoadFlowResult]
      */
     default CompletableFuture<LoadFlowResult> run(Network network, ComputationManager computationManager, String workingVariantId, LoadFlowParameters parameters, Reporter reporter) {
         return run(network, computationManager, workingVariantId, parameters);
     }
 
+    // Defines the JSON serializer/deserializer to use
+    Optional<JsonLoadFlowParameters.ExtensionSerializer<? extends Extension<LoadFlowParameters>>> getParametersExtensionSerializer();
+
+    // Load from platform config : will probably share code with the one above
+    <E extends Extension<LoadFlowParameters>> Optional<E> loadSpecificParameters(PlatformConfig config);
+
+    <E extends Extension<LoadFlowParameters>> Optional<Class<E>> getExtensionClass();
```

Comment:
The only purpose of this method is to better control the class of the extension. Would it be enough to just rely on extension.getClass() ? Is there a use case for have adding an extension with a different class (base one?) than the extension.getClass() one?